pkg/kds/util/resource.go

```diff
@@ -5,20 +5,22 @@ import (
 
 	envoy "github.com/envoyproxy/go-control-plane/envoy/api/v2"
 	envoy_types "github.com/envoyproxy/go-control-plane/pkg/cache/types"
-	"github.com/golang/protobuf/ptypes"
+	proto2 "github.com/golang/protobuf/proto"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
 
 	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
 	system_proto "github.com/kumahq/kuma/api/system/v1alpha1"
 	"github.com/kumahq/kuma/pkg/core/resources/model"
 	"github.com/kumahq/kuma/pkg/core/resources/registry"
-	"github.com/kumahq/kuma/pkg/util/proto"
+	util_proto "github.com/kumahq/kuma/pkg/util/proto"
 )
 
 func ToCoreResourceList(response *envoy.DiscoveryResponse) (model.ResourceList, error) {
 	krs := []*mesh_proto.KumaResource{}
 	for _, r := range response.Resources {
 		kr := &mesh_proto.KumaResource{}
-		if err := ptypes.UnmarshalAny(r, kr); err != nil {
+		if err := anypb.UnmarshalTo(r, kr, proto.UnmarshalOptions{}); err != nil {
```

Comment:
Consider wrapping this in `util_proto.UnmarshalAnyTo()` as a convenience API in the `MarshalAny` family. You can also add a `UnmarshalAnyProtoTo` or `UnmarshalAnyToV2` to wrap the proto message conversion.

Reply:
For me it's kind of questionable to introduce new util just to get rid of `, proto.UnmarshalOptions{}`, but technically it may be easier in the future to change a setting in UnmarshalOptions and apply on all of unmarshal operations.

I introduced MarshalAnyDeterministic because it was hard to do it in the previous API, now it is just an option in MarshalOptions.